modules/imapauth/lib/Auth/Source/MyAuth.php

```diff
@@ -0,0 +1,173 @@
+<?php
+class sspmod_imapauth_Auth_Source_MyAuth extends sspmod_core_Auth_UserPassBase {
```

Comment:
If you want, you can also use namespaces in your module, so that you don't need to give classes names that long. I also recommend to give every class a meaningful name. This could be, for example:

```php
<?php

namespace SimpleSAML\Module\imapauth\Auth\Source;

class IMAP extends sspmod_core_Auth_UserPassBase 
{
    ...
```

(we are slowly migrating all the classes to namespaces, so that means many of them are not there yet, like `sspmod_core_Auth_UserPassBase`)